### PR TITLE
Add doi metatag support

### PR DESCRIPTION
**Pull request by brinxmat**
Adds support for the Meta tag name "DOI"

### CODE REVIEW

File: metadata-transformer/src/main/java/no/unit/nva/metadata/type/Bibo.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package no.unit.nva.metadata;
+package no.unit.nva.metadata.type;
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/type/Bibo.java:1]: Moved the subpackage for tidiness, if you have a better name, tell me

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 2]
```diff
@@ -157,11 +158,19 @@ private boolean isPotentialDoiProperty(OntologyProperty ontologyProperty) {
...
         String property = statement.getPredicate().getLocalName();
-        Optional<Citation> citationValue = Citation.getByProperty(property);
+        Optional<Citation> citationValue = Citation.getTagByString(property);
         Optional<DcTerms> dcTermsValue = DcTerms.getTermByValue(property);
-        return citationValue.isPresent()
-                ? citationValue.get().getMappedTerm()
-                : dcTermsValue.orElse(null);
+        Optional<RawMetaTag> rawMetaTag = RawMetaTag.getTagByString(property);
+
+        OntologyProperty ontologyProperty = null;
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:165]: There's probably a tidier way of doing this

File: metadata-transformer/src/main/java/no/unit/nva/metadata/type/MetaTagSet.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package no.unit.nva.metadata.type;
+
+
+public interface MetaTagSet {
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/type/MetaTagSet.java:4]: Added interface to ensure that we don't extend and keep making naming issues

**Review by axthosarouris — APPROVED** [review]

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 4]
```diff
@@ -157,11 +158,19 @@ private boolean isPotentialDoiProperty(OntologyProperty ontologyProperty) {
 
     private OntologyProperty getMappedOntologyProperty(Statement statement) {
         String property = statement.getPredicate().getLocalName();
-        Optional<Citation> citationValue = Citation.getByProperty(property);
+        Optional<Citation> citationValue = Citation.getTagByString(property);
         Optional<DcTerms> dcTermsValue = DcTerms.getTermByValue(property);
```
Comment (mattiahj) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:162]: You've started calling this string tagName elsewhere, the argument property should probably be changed to reflect that.

Also, in general, the Citation and DcTerms classes are basically the same; certainly the methods getTagByString and getTermByValue are. Having them named the same would be nice, but perhaps we should go a step further and either make a general class for the both of them, or have a general MetaTag (or whatever) for what these classes have in common and then extend this class for each of these metadata tag types.

**Review by mattiahj — APPROVED** [review]

**Pull request merged**